libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/OfflineTiles.kt

```diff
@@ -0,0 +1,104 @@
+package com.mapbox.services.android.navigation.v5.navigation
+
+import com.mapbox.api.routetiles.v1.MapboxRouteTiles
+import com.mapbox.core.exceptions.ServicesException
+import com.mapbox.geojson.BoundingBox
+import okhttp3.ResponseBody
+import retrofit2.Callback
+
+/**
+ * This is a wrapper class for the [MapboxRouteTiles] class. This class takes care of
+ * interfacing with [MapboxRouteTiles] and receives a TAR file wrapped in a ResponseBody
+ * which can be handled using a
+ * [com.mapbox.services.android.navigation.v5.utils.DownloadTask].
+ */
+class OfflineTiles private constructor(
+    private val mapboxRouteTiles: MapboxRouteTiles,
+    private val version: String
+) {
+
+    /**
+     * Call when you have constructed your OfflineTiles object with your desired parameters.
+     * [Callback] must be passed into the method to handle both the response and failure.
+     *
+     * @param callback for retrofit
+     */
+    // FIXME make internal after kotlin migration
+    protected fun fetchRouteTiles(callback: Callback<ResponseBody>) {
+        mapboxRouteTiles.enqueueCall(callback)
+    }
+
+    /**
+     * Returns the version of the current builder.
+     *
+     * @return the version of the current builder
+     */
+    fun version(): String {
+        return version
+    }
+
+    /**
+     * This builder is used to create a new request to the Mapbox Route Tiles API. A request and
+     * therefore a builder must include a version, access token, and a [BoundingBox].
+     */
+    class Builder constructor() {
+        private var mapboxRouteTilesBuilder: MapboxRouteTiles.Builder = MapboxRouteTiles.builder()
+        private lateinit var _version: String
```

Comment:
Should we refactor this `lateinit var`? Seen https://github.com/mapbox/mapbox-navigation-android/pull/2080/files#diff-ce478a5a967cb29334078f0722adf81fR98 but not sure if we could avoid the `check` and `lateinit` somehow.

Reply:
We could make it `nullable`, but we definitely have to have check that `version` is not null when `build()` is called. 
IMO it looks good

Reply:
Yeah I was flagging because `lateinit` / `lazy` may cause unexpected NPEs if not handled properly 😉 You got it!